cppapi/client/filedatabase.cpp

```diff
@@ -1642,11 +1665,13 @@ CORBA::Any*   FileDatabase :: DbGetClassProperty(CORBA::Any& send)
 
 				if (m == nb_prop_defined)
 				{
+					// The requested property does not exist in the specified class
 					seq_length = seq_length + 2;
 					data_out->length(seq_length);
-					(*data_out)[index] = Tango::string_dup((*data_in)[i + 1].in());index++;
+					// The requested property name is returned, followed by a 0,
+					// meaning the length of this property value is 0 ( <=> class property not found )
+					(*data_out)[index] = Tango::string_dup((*data_in)[j].in());index++;
```

Comment:
This `index++` scattered all over the place is not only ugly but dangerous if one forgets to add it. There are few options that we could use to avoid that, at least these come to my mind:

1. Have a lambda that increases this index for us. But this could only be done in 9.4.
    ```cpp
    auto const push_data_out = [index = 0u](auto const &elem) mutable
    {
        (*data_out)[index] = Tango::string_dup(elem);
        index++;
    };
    ```
    Here we could simply call:
    ```cpp
    push_data_out(prop_name);
    push_data_out("0");
    ```

2. Collect all data in a vector and then copy that into the sequence. Note that we want to copy only the pointers. And it's not exception-safe but this is non-throwing code.
    ```cpp
    std::vector<char*> output{};
    output.push_back(Tango::string_dup("whatever"));

    auto *result = new DevVarStringArray{};
    result.length(output.size());
    std::memcpy(result.get_buffer(), output.data(), output.size() * sizeof(char*));
    ```

3. Write some utility classes which will provide a sane API for CORBA sequences. Maybe push_back like vector has, iteration support like in std;:span, etc. But this is something that we may rather consider for 9.4, not for backports.